app/assets/javascripts/application/messages.js

```diff
@@ -0,0 +1,96 @@
+var mumuki = mumuki || {};
+
+(function (mumuki) {
+
+    MessagesService = {
+        $body: function () {
+            return $('body')
+        },
+        $newMessageModal: function () {
+            return $('.new-message-modal');
+        },
+        $newMessageModalComponents: function () {
+            return MessagesService.$newMessageModal().find('.modal-body, .modal-footer');
+        },
+        collapseNewMessageModal: function () {
+            MessagesService.$newMessageModalComponents().toggleClass('hidden');
+        },
+        token: new mumuki.CsrfToken(),
+        setMessages: function (data) {
+            $('.badge-notifications').html(data.messages_count);
+            $('.notifications-box').toggleClass('notifications-box-empty', !data.has_messages);
+            $('.pending-messages-filter').removeClass('pending-messages-filter');
+            $('button.btn-submit').removeClass('disabled');
+            $('.pending-messages-text').remove()
+        },
+        readMessages: function (url) {
+            MessagesService.tokenRequest({
+                url: url,
+                method: 'POST',
+                success: MessagesService.setMessages,
+                xhrFields: {withCredentials: true}
+            })
+        },
+        tokenRequest: function (data) {
```

Comment:
Why not just `request`, or `doRequest`? I want to hide to the sender of this message that we are using tokens. More abstraction, more happiness :smile: